### ISSUE TITLE
Incompatible with python 3.8

### ISSUE THREAD

**Issue by jeblair** [kislyuk/rehash#4]
Something about the internals rehash uses seems to have changed in python 3.8.0.  This is using the "python:slim" docker image:

```
Python 3.8.0 (default, Oct 17 2019, 05:48:34) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import rehash, pickle
>>> hasher = rehash.sha256()
>>> pickle.dumps(hasher)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/rehash/__init__.py", line 52, in __getstate__
    ctx = self._get_evp_md_ctx()
  File "/usr/local/lib/python3.8/site-packages/rehash/__init__.py", line 46, in _get_evp_md_ctx
    if hasattr(c_evp_obj.contents.ctx, "contents"):
ValueError: NULL pointer access
```

**Comment by kislyuk** [comment]
Thank you for the report.

This appears to be the culprit: https://github.com/python/cpython/commit/5a4f82f457049b5b07b6fba4ca42bc1ecf597976#diff-80bbc723e944ebedb3f450b876b42056L33 - upstream removed a field from the EVPobject struct.

Before I can fix it, I'll need to migrate CI to GitHub Actions.

**Comment by kislyuk** [comment]
I migrated CI to github actions, but they don't support Python 3.8 yet: https://github.com/kislyuk/rehash/runs/269186836 - blocked on https://github.com/actions/setup-python/issues/30 for now.

**Comment by kislyuk** [comment]
Fixed in v1.0.0., please test.
